bodo/pandas/_pipeline.cpp

```diff
@@ -2,36 +2,69 @@
 
 #include "physical/result_collector.h"
 
+/*
+ * This has to be a recursive routine.  Each operator in the pipeline could
+ * say that it HAVE_MORE_OUTPUT in which case we need to call it again for the
+ * same input batch (or an empty input batch) once we've call the subsequent
```

Comment:
I think using same input batch can be dangerous depending on operator behavior and empty input batch is safer.

Reply:
From what I can tell, duckdb passes the same batch. src/parallel/pipeline_executor.cpp

`			// Operator(s) in the pipeline have returned `HAVE_MORE_OUTPUT` in the last Execute call
			// the operators have to be called with the same input chunk to produce the rest of the output
			D_ASSERT(source_chunk.size() > 0);
			result = ExecutePushInternal(source_chunk, chunk_budget);`